lib/rule-context.js

```diff
@@ -61,6 +62,71 @@ const PASSTHROUGHS = [
 //------------------------------------------------------------------------------
 
 /**
+ * Compares items in a fixes array by range.
+ * @param {Fix} a The first message.
+ * @param {Fix} b The second message.
+ * @returns {int} -1 if a comes before b, 1 if a comes after b, 0 if equal.
+ * @private
+ */
+function compareFixesByRange(a, b) {
+    return a.range[0] - b.range[0] || a.range[1] - b.range[1];
+}
+
+/**
+ * Merges the given fixes array into one.
+ * @param {Fix[]} fixes The fixes to merge.
+ * @param {SourceCode} sourceCode The source code object to get the text between fixes.
+ * @returns {void}
+ */
+function mergeFixes(fixes, sourceCode) {
```

Comment:
Worth noting that `mergeFixes` could be made an internal utility instead of a public API if there's any controversy around this API addition.